### PR TITLE
Testsuite: fix base/timer

### LINKED ISSUE #4455: base/timer unstable
In reference to #4439 

https://cdash.kyomu.43-1.org/testDetails.php?test=25150263&build=7648

### PR DESCRIPTION
**Pull request by tamiko**
- comparing concurrent timer runs on SMP machines with overclocking is
   useless

 - test the whole timer interface for consistency instead

Closes #4455

### CODE REVIEW

**Comment by drwells** [comment]
Is there any advantage to using `burn` instead of `std::this_thread::sleep_for`?

**Comment by tamiko** [comment]
@drwells It burns some CPU cycles so that we get larger time values?

File: tests/base/timer.cc [hunk 1]
```diff
@@ -62,24 +43,62 @@ int main ()
...
+  else
+    deallog << "ERROR - s2 should be greater than s1" << std::endl;
+
+  burn(50);
+  double s3 = t();
+
+  if (s3 == s2)
+    deallog << "OK" << std::endl;
+  else
+    deallog << "ERROR - s3 should be euqal to s2" << std::endl;
```
Comment (drwells) [inline tests/base/timer.cc]: should be `equal`

**Comment by drwells** [comment]
Fair enough. Please fix the typo and merge yourself.

**Pull request merged**